src/main/webapp/script.js

```diff
@@ -12,7 +12,118 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+/**
+ * Submits a fetch request to the /data url. Upon receiving a JSON encoding of the
+ * nodes and edges of the graph, renders the graph in a container on the page using
+ * the cytoscape.js library
+ */
+async function generateGraph() {
+  // Arrays to store the cytoscape graph node and edge objects
+  let graphNodes = [];
+  let graphEdges = [];
+  const response = await fetch("/data");
 
-function testFunc() {
-  fetch("/data").then(response => true);
-}
\ No newline at end of file
+  const serverErrorStatus = response.headers.get("serverError");
+
+  // Error on server side
+  if (serverErrorStatus !== null) {
+    displayError(serverErrorStatus);
+  }
+
+  const jsonResponse = await response.json();
+  // Graph nodes and edges received from server
+  let nodes = jsonResponse[0];
+  let edges = jsonResponse[1];
+
+  if (nodes && edges) {
+    // Add node to array of cytoscape nodes
+    nodes.forEach(node =>
+      graphNodes.push({
+        group: "nodes",
+        data: { id: node["name"] }
+      }))
+    // and edge to array of cytoscape edges
+    edges.forEach(edge => {
+      let start = edge["nodeU"]["name"];
+      let end = edge["nodeV"]["name"];
+      graphEdges.push({
+        group: "edges",
+        data: {
+          id: `edge${start}${end}`,
+          target: end,
+          source: start
+        }
+      });
+    })
+    getGraphDisplay(graphNodes, graphEdges);
+    return;
```

Comment:
Instead of 
```
if (valid) {
  // do valid stuff
  return;
}
displayError
```
It is more straightforward if you do 
```
if (invalid) {
   displayError
   return;
}
// do valid stuff
```